server.js

```diff
@@ -1,35 +1,308 @@
+      data: tracks,
+      success: true
+    })
+    } catch(err) {
+      res.status(400).json({
+        error: "Invalid request",
+        success: false
+      })
+    }
+})
+
+
+// get all tracks with high-tempo, ie. tracks with bpm >= 120
+app.get("/tracks/high-tempo", async (req, res) => {
+  try {
+    const tracksWithHighTempo = await Track.find({
+      bpm: { $gte: 120 }
+    })
+    res.status(200).json({
+      data: tracksWithHighTempo,
+      success: true
+    })
+  } catch(err) {
+    res.status(400).json({
+      error: "Invalid request",
+      success: false
+    })
+  }
+})
+
+
+// get all danceable tracks, ie tracks with danceability >= 60
+app.get("/tracks/danceable", async (req, res) => {
+  try {
+    const danceableTracks = await Track.find({
+      danceability: { $gte: 60 }
+    })
+    res.status(200).json({
+      data: danceableTracks,
+      success: true
+    })
+  } catch(err) {
+    res.status(400).json({
+      error: "Invalid request",
+      success: false
+    })
+  }
+})
+
+
+// get tracks by genre
+app.get("/tracks/genres/:genre", async (req, res) => {
+  try {
+    const { genre } = req.params
+    const tracksByGenre = await Track.find({
+      genre: new RegExp(genre, "i")
+    })
+    if (tracksByGenre) {
+      res.status(200).json({
+        data: tracksByGenre,
+        success: true
+      })
+    } else {
+      res.status(404).json({
+        data: "Sorry, genre not found",
+        success: false
+      })
+    }
+  } catch(err) {
+    res.status(400).json({
+      error: "Invalid request",
+      success: false
+    })
+  }
+})
+
+
+// get tracks by artist name
+app.get("/tracks/artists/:artist", async (req, res) => {
+  try {
+    const { artist } = req.params
+    const tracksByArtist = await Track.find({
+      artistName: new RegExp(artist, "i")
+    })
+    if (tracksByArtist) {
+      res.status(200).json({
+        data: tracksByArtist,
+        success: true
+      })
+    } else {
+      res.status(404).json({
+        data: "Sorry, artist not found",
+        success: false
+      })
+    }
+  } catch(err) {
+    res.status(400).json({
+      error: "Invalid request",
+      success: false
+    })
+  }
+})
+
+
+// get tracks by songtitle
+app.get("/tracks/songtitles/:songtitle", async (req, res) => {
+  try {
+    const { songtitle } = req.params
+    const singleTrack = await Track.findOne({
+      trackName: new RegExp(songtitle, "i")
+    })
+    if (singleTrack) {
+      res.status(200).json({
+        data: singleTrack,
+        success: true
+      })
+    } else {
+      res.status(404).json({
+        data: "Sorry, song not found",
+        success: false
+      })
+    }
+  } catch(err) {
+    res.status(400).json({
+      error: "Invalid request",
+      success: false
+    })
+  }
+})
+
+
+// get a single track based on database id
+app.get("/tracks/id/:id", async (req, res) => {
+  try {
+    const trackById = await Track.findById(req.params.id)
+    if (trackById) {
```

Comment:
In last weeks mongo api common pitfalls doc they want us to write endpoints like this:
_When you are accessing a single document with the key, use syntax like this: /shows/:id
rather than /shows/id/:id_